src/server/project.ts

```diff
@@ -715,6 +712,26 @@ namespace ts.server {
             }
         })();
 
+        private _isJsInferredProject = false;
+        set isJsInferredProject(newValue: boolean) {
+            if (newValue && !this._isJsInferredProject) {
+                this.setCompilerOptions(this.getCompilerOptions());

```

Comment:
also when before calling `setCompilerOptions` `this._isJsInferredProject` is not yet set so I'm not sure how control flow should enter `if` statement below
```
+            if (this._isJsInferredProject && typeof newOptions.maxNodeModuleJsDepth !== "number") {
 +                newOptions.maxNodeModuleJsDepth = 2;
 +            }
```